app/routes.py

```diff
@@ -1 +1,203 @@
+
+    return jsonify({"details": f'Task {task_to_delete.task_id} "{task_to_delete.title}" successfully deleted'}), 200
+
+@task_bp.route('/<task_id>/<mark_tf>', methods=['PATCH'])
+def change_value_of_task(task_id, mark_tf):
+    task_to_patch = get_model_from_id(Task, task_id)
+
+    try:
+        if mark_tf == 'mark_complete':
+            task_to_patch.is_complete = True # True or False
+            task_to_patch.completed_at = datetime.utcnow()
+            slackbot(task_to_patch.title)
+        elif mark_tf == 'mark_incomplete':
+            task_to_patch.is_complete = False # True or False
+            task_to_patch.completed_at = None
+    except KeyError:
+        return jsonify({"msg": "Missing needed data"}), 400 
+
+    db.session.commit()
+    return jsonify({"task": task_to_patch.to_dict()}), 200
+
+
+# goal
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goal_bp.route('', methods=['GET'])
+def get_all_goals():
+    goals = Goal.query.all()
+
+    result = []
+    for item in goals:
+        result.append(item.to_dict())
+
+    return jsonify(result), 200
+
+@goal_bp.route('/<goal_id>', methods=['GET'])
+def get_one_goal(goal_id):
+    chosen_goal = get_model_from_id(Goal, goal_id)
+    return jsonify({"goal": chosen_goal.to_dict()}), 200
+
+@goal_bp.route('', methods=['POST'])
+def create_one_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError:
+        return jsonify({"details": "Invalid data"}), 400
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({"goal": new_goal.to_dict()}), 201
+
+@goal_bp.route('/<goal_id>', methods=['PUT'])
+def update_one_goal(goal_id):
+    update_goal = get_model_from_id(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    try:
+        update_goal.title = request_body["title"]
+    except KeyError:
+        return jsonify({"msg": "Missing needed data"}), 400 
+
+    db.session.commit()
+    return jsonify({"goal": update_goal.to_dict()}), 200
+
+@goal_bp.route('/<goal_id>', methods=['DELETE'])
+def delete_one_goal(goal_id):
+    goal_to_delete = get_model_from_id(Goal, goal_id)
+
+    db.session.delete(goal_to_delete)
+    db.session.commit()
+
+    return jsonify({"details": f'Goal {goal_to_delete.goal_id} "{goal_to_delete.title}" successfully deleted'}), 200
+
+@goal_bp.route('/<goal_id>/tasks', methods=['GET'])
+def get_tasks_for_goal(goal_id):
+    chosen_goal = get_model_from_id(Goal, goal_id)
+
+    return jsonify(chosen_goal.to_dict_task()), 200
+
+@goal_bp.route('/<goal_id>/tasks', methods=['POST'])
+def create_tasks_for_goal(goal_id):
+    matching_goal = Goal.query.get(goal_id)
+
+    request_body = request.get_json()
+    task_ids = request_body["task_ids"]
+    
+    tasks = matching_goal.get_task_list()
+    goal_task_ids = [x["id"] for x in tasks]
+    for tid in task_ids: 
+        if tid not in goal_task_ids:
+            new_task = Task(
```

Comment:
We should not be creating new Tasks. We do not get to control what id the new Task gets. It is assumed that if we have a task_id given to us, it is for a Task that exists. We should check the database for the existing task instead of creating a new one. If the task does not exist, we can throw an error for the user.